server/athenian/api/openapi/openapi.yaml

```diff
@@ -1083,6 +1308,152 @@ components:
+      description: Team creation request.
+      example:
+        account: 1
+        name: Engineering
+        members:
+        - github.com/gkwillie
+        - github.com/se7entyse7en
+      properties:
+        account:
+          description: Account identifier. That account will own the created team.
+          type: integer
+        name:
+          description: Name of the team.
+          type: string
+        members:
+          description: Members of the team.
+          $ref: '#/components/schemas/DeveloperSet'
+      required:
+      - account
+      - members
+      - name
+      type: object
+    TeamUpdateRequest:
+      additionalProperties: false
+      description: Team update request.
+      example:
+        id: 1
+        name: Engineering
+        members:
+        - github.com/gkwillie
+        - github.com/se7entyse7en
+      properties:
+        id:
+          description: Team identifier.
+          type: integer
+        name:
+          description: Name of the team.
+          type: string
+        members:
+          description: Members of the team.
+          $ref: '#/components/schemas/DeveloperSet'
+      required:
+      - id
+      - members
+      - name
+      type: object
+    Team:
+      description: List of team members.
+      example:
+        name: Engineering
+        members:
+        - login: github.com/gkwillie
+          id: github|60340680
+          native_id: 60340680
+          name: Groundskeeper Willie
+          email: bot@athenian.co
+          picture: https://avatars0.githubusercontent.com/u/60340680?v=4
+          updated: 2020-01-23T12:00:00Z
+        - login: github.com/se7entyse7en
+          id: github|5599208
+          native_id: 5599208
+          name: Lou Marvin Caraig
+          email: loumarvincaraig@gmail.com
+          picture: https://avatars1.githubusercontent.com/u/5599208?v=4
+          updated: 2020-05-05T18:19:05Z
+      properties:
+        name:
+          description: Name of the team.
+          type: string
+        members:
+          items:
+            $ref: '#/components/schemas/User'
+          type: array
+      required:
+      - members
+      - name
+      type: object
+    TeamListItem:
+      description: Item of the list of teams.
+      example:
+      - id: 1
+        team:
+          name: Engineering
+          members:
+          - login: github.com/vmarkovtsev
+            id: github|2793551
+            native_id: 2793551
+            name: Vadim Markovtsev
+            email: vadim@athenian.co
+            picture: https://avatars3.githubusercontent.com/u/2793551?v=4
+            updated: 2020-05-01T17:15:51Z
+          - login: github.com/se7entyse7en
+            id: github|5599208
+            native_id: 5599208
+            name: Lou Marvin Caraig
+            email: waren@athenian.co
+            picture: https://avatars1.githubusercontent.com/u/5599208?v=4
+            updated: 2020-05-05T18:19:05Z
+      properties:
+        id:
```

Comment:
Should we move `id` inside the `Team`? All the places that I see have them together.

I am also worried about how the team listing will work. If we accept a list of github logins then we will not always have an `id` - that's an Auth0 user ID. We cannot make it optional for the `User` either. It looks like we have to return back our `TeamMember`.

Reply:
> Should we move id inside the Team? All the places that I see have them together.

Good point, indeed it was needed when the put needed to handle empty id. 👍 

> I am also worried about how the team listing will work. If we accept a list of github logins then we will not always have an id - that's an Auth0 user ID. We cannot make it optional for the User either. It looks like we have to return back our TeamMember.

Okay, I got the point. But this is related mostly to creation and update itself, isn't it? Instead of accepting a list of GH handles they should accept a list of ids.

Reply:
IDK what's the product decision here: allow any valid github login, allow any github login from the account's orgs or allow registered account users...

Reply:
I'd go strict, that is users that are member of the account, that would simplify a lot. Also in that case no need to go back to `TeamMember` since all members would be regular `User`s. \cc @warenlg